### DISCUSSION TITLE
Receiving an Unfamiliar ID Instead of Username with OAuth2 Azure EntraID Authentication

### DISCUSSION THREAD

**Discussion by AlakmarShafin** [trinodb/trino#23139] [Q&A]
I'm working on configuring Azure Entra ID with Trino for authentication. Although the setup appears to be successful, I'm encountering an issue where a strange ID is being displayed in the Trino UI instead of my Azure ID.
![image](https://github.com/user-attachments/assets/eda62653-26d5-412f-ad11-3d3ef47ff560)
I'm unsure where this ID originates from. I've verified that it isn't the object ID of the AAD user. Since I'm using Azure ID in file-based access control, I need to identify this ID to map it with the corresponding Azure ID.

### Below is my config.properties

```
coordinator=true
node-scheduler.include-coordinator=false
http-server.http.port=80
discovery.uri=http://localhost:80
http-server.https.enabled=true
http-server.https.port=443
http-server.https.keystore.path=/etc/trino/healthrxcert.pem
http-server.authentication.type=oauth2,PASSWORD
password-authenticator.config-files=/etc/trino/password-authenticator.properties
http-server.authentication.oauth2.issuer=https://login.microsoftonline.com/<tenent ID>/v2.0
http-server.authentication.oauth2.client-id=<ID>
http-server.authentication.oauth2.client-secret=<SECRET>
http-server.authentication.oauth2.auth-url=https://login.microsoftonline.com/<tenent ID>/oauth2/v2.0/authorize
http-server.authentication.oauth2.token-url=https://login.microsoftonline.com/<tenent ID>/oauth2/v2.0/token
http-server.authentication.oauth2.jwks-url=https://login.microsoftonline.com/<tenent ID>/discovery/v2.0/keys
http-server.authentication.oauth2.userinfo-url=https://graph.microsoft.com/oidc/userinfo
http-server.authentication.oauth2.scopes=openid
web-ui.authentication.type=oauth2
```

**Comment by hashhar** [comment]
Try inspecting the JWT token returned by Azure Entra. This is probably the `sub` value in the token. You can use a different field for the principal by setting `http-server.authentication.oauth2.principal-field`. See docs for more - https://trino.io/docs/current/security/oauth2.html#trino-server-configuration

**Comment by AlakmarShafin** [comment]
Hi @hashhar, thanks for replying.

 I have already tried this, when I configure this property other than "sub" (I have tried "email" and "upn" ) it gives me an error: "unauthorized" on UI after login using the same azure id which was previously working with "sub".
<img width="497" alt="image" src="https://github.com/user-attachments/assets/ff5b3dc9-a198-43fc-8fac-0bd516f8c730">

**Reply by hashhar** [reply]
Try with `http-server.authentication.oauth2.scopes=openid,offline` or `http-server.authentication.oauth2.scopes=openid,offline_access`.
